src/Hl7.Fhir.Serialization.Tests/RoundtripNewSerializers.cs

```diff
@@ -0,0 +1,220 @@
+            var jsonOptions = new JsonSerializerOptions().ForFhir(typeof(Patient).Assembly).Pretty();
+
+            files.ForEach(f => objects.Add(new object[] { f, xmlSerializer, xmlDeserializer, jsonOptions }));
+
+            return objects;
+        }
+
+        public static string GetTestDisplayNames(MethodInfo methodInfo, object[] values)
+        {
+            return Path.GetFileName((string)values[0]);
+        }
+
+        private static void doRoundTrip(string baseTestPath, string file, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            var errors = new List<string>();
+
+            var intermediate1Path = Path.Combine(baseTestPath, intermediate1Folder);
+
+            //First conversion
+            convertFile(file, intermediate1Path, xmlSerializer, xmlDeserializer, jsonOptions, errors);
+            var intermediate2Path = Path.Combine(baseTestPath, intermediate2Folder);
+
+            string exampleWithOppositeExtension = changeFileExtension(file);
+
+            string intermediate1File = Path.Combine(intermediate1Path, exampleWithOppositeExtension);
+            //Convert back to original
+            convertFile(intermediate1File, intermediate2Path, xmlSerializer, xmlDeserializer, jsonOptions, errors);
+
+            //compare original with roundtrip output
+            compareFile(file, Path.Combine(intermediate2Path, Path.GetFileName(file)), errors);
+            Assert.AreEqual(0, errors.Count, "Errors were encountered comparing converted content");
+        }
+
+        private static string changeFileExtension(string file)
+        {
+            string exampleName = Path.GetFileNameWithoutExtension(file);
+            string ext = Path.GetExtension(file);
+            var toExt = ext == ".xml" ? ".json" : ".xml";
+            return exampleName + toExt;
+        }
+
+        private static ZipArchive extractTestZip(string filename)
+        {
+            string file = getFullPathForExample(filename);
+            return ZipFile.OpenRead(file);
+        }
+
+        private static string getFullPathForExample(string filename) => Path.Combine("TestData", filename);
+
+        private static void createEmptyDir(string baseTestPath)
+        {
+            if (Directory.Exists(baseTestPath)) Directory.Delete(baseTestPath, true);
+            Directory.CreateDirectory(baseTestPath);
+        }
+
+        private static void convertFile(string file, string outputPath, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions, List<string> errors)
+        {
+            string exampleWithOppositeExtension = changeFileExtension(file);
+            string outputFile = Path.Combine(outputPath, exampleWithOppositeExtension);
+
+            if (!(file.Contains("expansions.") || file.Contains("profiles-resources") || file.Contains("profiles-others") || file.Contains("valuesets.")))
```

Comment:
Can we extract this to a method so we have it centralized in the top of this unit test?